assets/js/modules/analytics-4/utils/__fixtures__/mocked-pivot-report-multi-dimension-fieldnames.json

```diff
@@ -0,0 +1,437 @@
+{
+    "dimensionHeaders": [
+      {
+        "name": "city"
+      },
+      {
+        "name": "audienceResourceName"
+      },
+      {
+        "name": "country"
+      }
+    ],
+    "metricHeaders": [
+      {
+        "name": "totalUsers",
+        "type": "TYPE_INTEGER"
+      },
+      {
+        "name": "activeUsers",
+        "type": "TYPE_INTEGER"
+      }
+    ],
+    "pivotHeaders": [
+      {
+        "pivotDimensionHeaders": [
+          {
+            "dimensionValues": [
+              {
+                "value": "properties/12345/audiences/1"
+              }
+            ]
+          },
+          {
+            "dimensionValues": [
+              {
+                "value": "properties/12345/audiences/2"
+              }
+            ]
+          }
+        ],
+        "rowCount": 2
+      },
+      {
+        "pivotDimensionHeaders": [
+          {
+            "dimensionValues": [
+              {
+                "value": "Dublin"
+              }
+            ]
+          },
+          {
+            "dimensionValues": [
+              {
+                "value": "(not set)"
+              }
+            ]
+          },
+          {
+            "dimensionValues": [
+              {
+                "value": "Cork"
+              }
+            ]
+          }
+        ],
+        "rowCount": 3
+      },
+      {
+        "pivotDimensionHeaders": [
+          {
+            "dimensionValues": [
+              {
+                "value": "United States"
+              }
+            ]
+          },
+          {
+            "dimensionValues": [
+              {
+                "value": "United Kingdom"
+              }
+            ]
+          },
+          {
+            "dimensionValues": [
+              {
+                "value": "India"
+              }
+            ]
+          }
+        ],
+        "rowCount": 3
+      },
+      {
+        "pivotDimensionHeaders": [
+          {
+            "dimensionValues": [
+              {
+                "value": "United States"
+              }
+            ]
+          },
+          {
+            "dimensionValues": [
+              {
+                "value": "United Kingdom"
+              }
+            ]
+          },
+          {
+            "dimensionValues": [
+              {
+                "value": "India"
+              }
+            ]
+          }
+        ],
+        "rowCount": 3
+      }
+    ],
+    "aggregates": [
```

Comment:
The number and shape of the aggregates is also not correct. As you can see, for each pivot, we should have effectively have the number of rows (i.e. `limit` in our output) x 3 (TOTAL, MINIMUM and MAXIMUM).

So there should be 2x3 + 3x3 + 3x3 = 24 entries in `aggregates`. As you can see the current output only has 9 entries. Also, the each entry should contain 4 dimension values.

<details>
<summary>Expected shape (taken from my example)</summary>

```json
	"aggregates": [
		{
			"dimensionValues": [
				{
					"value": "new"
				},
				{
					"value": "RESERVED_TOTAL"
				},
				{
					"value": "RESERVED_TOTAL"
				},
				{
					"value": "RESERVED_TOTAL"
				}
			],
			"metricValues": [
				{
					"value": "4892"
				},
				{
					"value": "4892"
				}
			]
		},
		{
			"dimensionValues": [
				{
					"value": "RESERVED_TOTAL"
				},
				{
					"value": "RESERVED_TOTAL"
				},
				{
					"value": "RESERVED_TOTAL"
				},
				{
					"value": "iOS"
				}
			],
			"metricValues": [
				{
					"value": "2104"
				},
				{
					"value": "2104"
				}
			]
		},
		{
			"dimensionValues": [
				{
					"value": "RESERVED_TOTAL"
				},
				{
					"value": "RESERVED_TOTAL"
				},
				{
					"value": "RESERVED_TOTAL"
				},
				{
					"value": "Android"
				}
			],
			"metricValues": [
				{
					"value": "1936"
				},
				{
					"value": "1936"
				}
			]
		},
		{
			"dimensionValues": [
				{
					"value": "RESERVED_TOTAL"
				},
				{
					"value": "Dublin"
				},
				{
					"value": "Ireland"
				},
				{
					"value": "RESERVED_TOTAL"
				}
			],
			"metricValues": [
				{
					"value": "1202"
				},
				{
					"value": "1202"
				}
			]
		},
		{
			"dimensionValues": [
				{
					"value": "RESERVED_TOTAL"
				},
				{
					"value": "RESERVED_TOTAL"
				},
				{
					"value": "RESERVED_TOTAL"
				},
				{
					"value": "Windows"
				}
			],
			"metricValues": [
				{
					"value": "690"
				},
				{
					"value": "690"
				}
			]
		},
		{
			"dimensionValues": [
				{
					"value": "RESERVED_MAX"
				},
				{
					"value": "Dublin"
				},
				{
					"value": "Ireland"
				},
				{
					"value": "RESERVED_MAX"
				}
			],
			"metricValues": [
				{
					"value": "551"
				},
				{
					"value": "551"
				}
			]
		},
		{
			"dimensionValues": [
				{
					"value": "RESERVED_MAX"
				},
				{
					"value": "RESERVED_MAX"
				},
				{
					"value": "RESERVED_MAX"
				},
				{
					"value": "iOS"
				}
			],
			"metricValues": [
				{
					"value": "551"
				},
				{
					"value": "551"
				}
			]
		},
		{
			"dimensionValues": [
				{
					"value": "new"
				},
				{
					"value": "RESERVED_MAX"
				},
				{
					"value": "RESERVED_MAX"
				},
				{
					"value": "RESERVED_MAX"
				}
			],
			"metricValues": [
				{
					"value": "551"
				},
				{
					"value": "551"
				}
			]
		},
		{
			"dimensionValues": [
				{
					"value": "returning"
				},
				{
					"value": "RESERVED_TOTAL"
				},
				{
					"value": "RESERVED_TOTAL"
				},
				{
					"value": "RESERVED_TOTAL"
				}
			],
			"metricValues": [
				{
					"value": "530"
				},
				{
					"value": "530"
				}
			]
		},
		{
			"dimensionValues": [
				{
					"value": "RESERVED_MAX"
				},
				{
					"value": "RESERVED_MAX"
				},
				{
					"value": "RESERVED_MAX"
				},
				{
					"value": "Android"
				}
			],
			"metricValues": [
				{
					"value": "430"
				},
				{
					"value": "430"
				}
			]
		},
		{
			"dimensionValues": [
				{
					"value": "RESERVED_TOTAL"
				},
				{
					"value": "Cork"
				},
				{
					"value": "Ireland"
				},
				{
					"value": "RESERVED_TOTAL"
				}
			],
			"metricValues": [
				{
					"value": "244"
				},
				{
					"value": "244"
				}
			]
		},
		{
			"dimensionValues": [
				{
					"value": "RESERVED_TOTAL"
				},
				{
					"value": "Sao Paulo"
				},
				{
					"value": "Brazil"
				},
				{
					"value": "RESERVED_TOTAL"
				}
			],
			"metricValues": [
				{
					"value": "196"
				},
				{
					"value": "196"
				}
			]
		},
		{
			"dimensionValues": [
				{
					"value": "RESERVED_MAX"
				},
				{
					"value": "Cork"
				},
				{
					"value": "Ireland"
				},
				{
					"value": "RESERVED_MAX"
				}
			],
			"metricValues": [
				{
					"value": "191"
				},
				{
					"value": "191"
				}
			]
		},
		{
			"dimensionValues": [
				{
					"value": "RESERVED_MAX"
				},
				{
					"value": "RESERVED_MAX"
				},
				{
					"value": "RESERVED_MAX"
				},
				{
					"value": "Windows"
				}
			],
			"metricValues": [
				{
					"value": "121"
				},
				{
					"value": "121"
				}
			]
		},
		{
			"dimensionValues": [
				{
					"value": "RESERVED_MAX"
				},
				{
					"value": "Sao Paulo"
				},
				{
					"value": "Brazil"
				},
				{
					"value": "RESERVED_MAX"
				}
			],
			"metricValues": [
				{
					"value": "83"
				},
				{
					"value": "83"
				}
			]
		},
		{
			"dimensionValues": [
				{
					"value": "returning"
				},
				{
					"value": "RESERVED_MAX"
				},
				{
					"value": "RESERVED_MAX"
				},
				{
					"value": "RESERVED_MAX"
				}
			],
			"metricValues": [
				{
					"value": "78"
				},
				{
					"value": "78"
				}
			]
		},
		{
			"dimensionValues": [
				{
					"value": "RESERVED_MIN"
				},
				{
					"value": "Dublin"
				},
				{
					"value": "Ireland"
				},
				{
					"value": "RESERVED_MIN"
				}
			],
			"metricValues": [
				{
					"value": "4"
				},
				{
					"value": "4"
				}
			]
		},
		{
			"dimensionValues": [
				{
					"value": "RESERVED_MIN"
				},
				{
					"value": "Cork"
				},
				{
					"value": "Ireland"
				},
				{
					"value": "RESERVED_MIN"
				}
			],
			"metricValues": [
				{
					"value": "1"
				},
				{
					"value": "1"
				}
			]
		},
		{
			"dimensionValues": [
				{
					"value": "RESERVED_MIN"
				},
				{
					"value": "RESERVED_MIN"
				},
				{
					"value": "RESERVED_MIN"
				},
				{
					"value": "Android"
				}
			],
			"metricValues": [
				{
					"value": "1"
				},
				{
					"value": "1"
				}
			]
		},
		{
			"dimensionValues": [
				{
					"value": "RESERVED_MIN"
				},
				{
					"value": "RESERVED_MIN"
				},
				{
					"value": "RESERVED_MIN"
				},
				{
					"value": "Windows"
				}
			],
			"metricValues": [
				{
					"value": "1"
				},
				{
					"value": "1"
				}
			]
		},
		{
			"dimensionValues": [
				{
					"value": "RESERVED_MIN"
				},
				{
					"value": "RESERVED_MIN"
				},
				{
					"value": "RESERVED_MIN"
				},
				{
					"value": "iOS"
				}
			],
			"metricValues": [
				{
					"value": "1"
				},
				{
					"value": "1"
				}
			]
		},
		{
			"dimensionValues": [
				{
					"value": "RESERVED_MIN"
				},
				{
					"value": "Sao Paulo"
				},
				{
					"value": "Brazil"
				},
				{
					"value": "RESERVED_MIN"
				}
			],
			"metricValues": [
				{
					"value": "1"
				},
				{
					"value": "1"
				}
			]
		},
		{
			"dimensionValues": [
				{
					"value": "new"
				},
				{
					"value": "RESERVED_MIN"
				},
				{
					"value": "RESERVED_MIN"
				},
				{
					"value": "RESERVED_MIN"
				}
			],
			"metricValues": [
				{
					"value": "1"
				},
				{
					"value": "1"
				}
			]
		},
		{
			"dimensionValues": [
				{
					"value": "returning"
				},
				{
					"value": "RESERVED_MIN"
				},
				{
					"value": "RESERVED_MIN"
				},
				{
					"value": "RESERVED_MIN"
				}
			],
			"metricValues": [
				{
					"value": "1"
				},
				{
					"value": "1"
				}
			]
		}
	],
```

</details>

<details>
<summary>Current output</summary>

```json
    "aggregates": [
      {
        "dimensionValues": [
          {
            "value": "RESERVED_MIN"
          },
          {
            "value": "city"
          }
        ],
        "metricValues": [
          {
            "value": "27"
          },
          {
            "value": "91"
          }
        ]
      },
      {
        "dimensionValues": [
          {
            "value": "RESERVED_MIN"
          },
          {
            "value": "audienceResourceName"
          }
        ],
        "metricValues": [
          {
            "value": "86"
          },
          {
            "value": "53"
          }
        ]
      },
      {
        "dimensionValues": [
          {
            "value": "RESERVED_MIN"
          },
          {
            "value": "country"
          }
        ],
        "metricValues": [
          {
            "value": "38"
          },
          {
            "value": "71"
          }
        ]
      },
      {
        "dimensionValues": [
          {
            "value": "RESERVED_MAX"
          },
          {
            "value": "city"
          }
        ],
        "metricValues": [
          {
            "value": "21"
          },
          {
            "value": "74"
          }
        ]
      },
      {
        "dimensionValues": [
          {
            "value": "RESERVED_MAX"
          },
          {
            "value": "audienceResourceName"
          }
        ],
        "metricValues": [
          {
            "value": "66"
          },
          {
            "value": "80"
          }
        ]
      },
      {
        "dimensionValues": [
          {
            "value": "RESERVED_MAX"
          },
          {
            "value": "country"
          }
        ],
        "metricValues": [
          {
            "value": "16"
          },
          {
            "value": "22"
          }
        ]
      },
      {
        "dimensionValues": [
          {
            "value": "RESERVED_TOTAL"
          },
          {
            "value": "city"
          }
        ],
        "metricValues": [
          {
            "value": "45"
          },
          {
            "value": "34"
          }
        ]
      },
      {
        "dimensionValues": [
          {
            "value": "RESERVED_TOTAL"
          },
          {
            "value": "audienceResourceName"
          }
        ],
        "metricValues": [
          {
            "value": "23"
          },
          {
            "value": "91"
          }
        ]
      },
      {
        "dimensionValues": [
          {
            "value": "RESERVED_TOTAL"
          },
          {
            "value": "country"
          }
        ],
        "metricValues": [
          {
            "value": "38"
          },
          {
            "value": "66"
          }
        ]
      }
    ],
```